### ISSUE TITLE
PostgreSQL datasource include_schema_name bug

### ISSUE THREAD

**Issue by lmcewen-helix** [fivetran/great_expectations#6260]
The update from 0.15.21 to 0.15.22 seems to have introduced a bug related to the `include_schema_name` parameter in the great_expectations.yml under the InferredAssetSqlDataConnector datasources. The error we get is:   

```
File ".../great_expectations/datasource/data_connector/configured_asset_sql_data_connector.py", line 196, in get_batch_definition_list_from_batch_request
    f"data_asset_name {batch_request.data_asset_name} is not recognized."
```

The data asset is a table within a postgreSQL database schema, formatted as `[schema].[table]`. The data_asset_name within the batch_request listed in the checkpoint file is not recognized. 

Datasources snippet:
```
datasources: 
  target:
    data_connectors:
      default_runtime_data_connector_name:
        class_name: RuntimeDataConnector
        batch_identifiers:
          - default_identifier_name
        module_name: great_expectations.datasource.data_connector
      default_inferred_data_connector_name:
        include_schema_name: true
        class_name: InferredAssetSqlDataConnector
        module_name: great_expectations.datasource.data_connector
    class_name: Datasource
    module_name: great_expectations.datasource
    execution_engine:
      class_name: SqlAlchemyExecutionEngine
      module_name: great_expectations.execution_engine
      credentials:
        host: [host]
        port: '5432'
        username: [username]
        password: [password]
        database: [database]
        drivername: postgresql
```

Batch request snippet from checkpoint:
```
- batch_request:
      datasource_name: target
      data_connector_name: default_inferred_data_connector_name
      data_asset_name: myschema.mytable
      data_connector_query:
        index: -1
```

**To Reproduce**
1. Set up postgres environment and indicate data source parameters as above including the `include_schema_name: true`
2. Have the data_asset_name: [myschema].[mytable] in the batch request

**Expected behavior**
The data assets to be found and the checkpoints to run properly

**Environment**
 - Operating System: MacOS
 - Great Expectations Version: 0.15.22

**Note:** I've found that if you remove both the `include_schema_name: true` and the schema part of the data_asset_name (just have the table name) in the batch request, it seems to resolve. 

I have already discussed this error with @alexsherstinsky and, after going through it together, he asked that I post the issue here. Let me know if I can provide any more details that would help. Thank you very much!

An issue earlier this month that looks like it may be similar: https://github.com/great-expectations/great_expectations/issues/6140

A current bug fix by @NathanFarmer may already be working on this? https://github.com/great-expectations/great_expectations/pull/6166

**Comment by NathanFarmer** [comment]
@lmcewen-helix Yes, I can confirm that I am working on a fix for this.

**Comment by NathanFarmer** [comment]
This bug has been fixed and will be included as part of release 0.15.30.

**Comment by lmcewen-helix** [comment]
Thank you!
